test-programs/compressed-token-test/tests/test.rs

```diff
@@ -243,27 +248,424 @@ async fn test_mint_to<const MINTS: usize, const ITER: usize>() {
+        amounts: amounts.clone(),
+        public_keys: recipients.clone(),
+    };
+
+    // 1. Try to mint token from `mint_1` and sign the transaction with `mint_2`
+    //    authority.
+    {
+        let instruction = create_mint_to_instruction(
+            &payer_2.pubkey(),
+            &payer_2.pubkey(),
+            &mint_1,
+            &merkle_tree_pubkey,
+            amounts.clone(),
+            recipients.clone(),
+        );
+        let result = rpc
+            .create_and_send_transaction(&[instruction], &payer_2.pubkey(), &[&payer_2])
+            .await;
+        assert_rpc_error(
+            result,
+            0,
+            light_compressed_token::ErrorCode::InvalidAuthorityMint.into(),
+        )
+        .unwrap();
+    }
+    // 2. Try to mint token from `mint_2` and sign the transaction with `mint_1`
+    //    authority.
+    {
+        let instruction = create_mint_to_instruction(
+            &payer_1.pubkey(),
+            &payer_1.pubkey(),
+            &mint_2,
+            &merkle_tree_pubkey,
+            amounts.clone(),
+            recipients.clone(),
+        );
+        let result = rpc
+            .create_and_send_transaction(&[instruction], &payer_1.pubkey(), &[&payer_1])
+            .await;
+        assert_rpc_error(
+            result,
+            0,
+            light_compressed_token::ErrorCode::InvalidAuthorityMint.into(),
+        )
+        .unwrap();
+    }
+    // 3. Try to mint token from `mint_1` while using `mint_2` pool.
+    {
+        let accounts = light_compressed_token::accounts::MintToInstruction {
+            fee_payer: payer_1.pubkey(),
+            authority: payer_1.pubkey(),
+            cpi_authority_pda: get_cpi_authority_pda().0,
+            mint: mint_1.clone(),
+            token_pool_pda: mint_pool_2,
+            token_program: anchor_spl::token::ID,
+            light_system_program: light_system_program::ID,
+            registered_program_pda: light_system_program::utils::get_registered_program_pda(
+                &light_system_program::ID,
+            ),
+            noop_program: Pubkey::new_from_array(
+                account_compression::utils::constants::NOOP_PUBKEY,
+            ),
+            account_compression_authority: light_system_program::utils::get_cpi_authority_pda(
+                &light_system_program::ID,
+            ),
+            account_compression_program: account_compression::ID,
+            merkle_tree: merkle_tree_pubkey,
+            self_program: light_compressed_token::ID,
+            system_program: system_program::ID,
+        };
+        let instruction = Instruction {
+            program_id: light_compressed_token::ID,
+            accounts: accounts.to_account_metas(Some(true)),
+            data: instruction_data.data(),
+        };
+        rpc.create_and_send_transaction(&[instruction], &payer_1.pubkey(), &[&payer_1])
+            .await
+            .unwrap_err();
```

Comment:
check for correct error pls
``
hread 'test_mint_to_failing' panicked at test-programs/compressed-token-test/tests/test.rs:423:14:
called `Result::unwrap()` on an `Err` value: TransactionError(InstructionError(0, Custom(3)))
``

Reply:
The reason I didn't do it before is that `spl_token::error::TokenError` doesn't implement From/Into `u32`, so calling `into()` on its variants doesn't work.

But I got it working with `as u32`:

```rust
assert_rpc_error(result, 0, TokenError::MintMismatch as u32).unwrap();
```

This trick will work only for enums which don't carry data. Should work for the most of enums from solana libs, but some of them (like `InstructionError`), don't work with this trick.